docs/components/config.rst

```diff
@@ -1,6 +1,177 @@
 Configuration
 #############
 
+For basic information about configuring your Plugin, see :ref:`Config file`. This document goes into more detail by explaining how to add User-configurable options to Mautic.
 
 Configuration parameters
-************************
\ No newline at end of file
+************************
+
+You can find Mautic's configuration in ``app/config/local.php``. Plugins can leverage custom config parameters to use within its code.
+
+You can define every configuration option you need. Make sure it has a default set in the :ref:`Plugin's config file<Config file>`.
+This prevents Symfony from throwing errors if the parameter gets used during cache compilation or if accessed directly from the container without checking if it exists first.
+Defining the parameters in the Plugin's config file ensures that it always exists.
+
+To add config options to the Configuration section, you will need to add an :ref:`Event Listener<Event listeners>`, a config  :ref:`form type<Forms>`, and a specific view (TODO).
```

Comment:
This needs #86 to be merged first, so we can link to the Form Type section from here.

Reply:
That PR is merged now so should be good to go with this comment :)